pcdsdevices/lodcm.py

```diff
@@ -1137,9 +1131,124 @@ def calc_energy(self, energy, material=None, reflection=None):
         """
         # try to determine the material and reflection:
         material = material or self.get_material()
-        reflection = reflection or self.get_reflection(
-            as_tuple=True, check=True)
-        return self.energy.calc_energy(energy, material, reflection)
+        reflection = reflection or self.get_reflection()
+        return self.energy.calc_geometry(energy, material, reflection)
+
+    def set_energy(self, energy, material=None, reflection=None):
+        """
+        Set the current positions of th1, th2, z1 and z2 offset motors.
+
+        Parameters
+        ----------
+        energy : number
+            Energy in keV.
+        material : str, optional
+            Crystal material. E.g.: `Si`
+        reflection : tuple, optional
+            Crystal's reflections. E.g.: `(1,1,1)`
+        """
+        # try to determine the material and reflection:
+        material = material or self.get_material()
+        reflection = reflection or self.get_reflection()
+        th, z = diffraction.get_lom_geometry(energy * 1e3, material,
+                                             reflection)
+        # TODO: probably not the correct approach here to use
+        # set_current_position
+        if material == 'Si':
+            self.th1Si.set_current_position(th)
+            self.th2Si.set_current_position(th)
+            self.z1Si.set_current_position(-z)
+            self.z2Si.set_current_position(z)
+        elif material == 'C':
+            self.th1C.set_current_position(th)
+            self.th2C.set_current_position(th)
+            self.z1C.set_current_position(-z)
+            self.z2C.set_current_position(z)
+        else:
+            raise ValueError("Invalid material: %s", material)
+
+    def tweak_x(self, x_value, material=None, wait=False):
```

Comment:
Should I be changing the naming here as well to reflect exactly what we have in the old code?
instead of `tweak_x` -> `tweakX`?

and the other ones?
`tweeak_parallel` - `tweakParallel`
`set_energy` - `setE`
.....
others?

I think it would not be too hard to get used to the new style, but it is easy for me to say, I am not the one using these....

Reply:
I don't have any insight here